components/brave_today/browser/brave_news_controller.cc

```diff
@@ -213,38 +215,41 @@ void BraveNewsController::GetImageData(const GURL& padded_image_url,
     std::move(callback).Run(std::move(args));
     return;
   }
-  // Handler url download response
+  // Use file ending to determine if response
+  // will contain (Brave's PrivateCDN) padding or
+  // be a direct image
   const auto file_name = padded_image_url.path();
   const std::string ending = ".pad";
   const bool is_padded =
       (file_name.compare(file_name.length() - ending.length(), ending.length(),
                          ending) == 0);
   VLOG(3) << "is padded: " << is_padded;
-  auto onPaddedImageResponse = base::BindOnce(
-      [](GetImageDataCallback callback, const bool is_padded, const int status,
-         const std::string& body,
-         const base::flat_map<std::string, std::string>& headers) {
-        // Attempt to remove byte padding if applicable
-        base::StringPiece body_payload(body.data(), body.size());
-        if (status < 200 || status >= 300 ||
-            (is_padded &&
-             !brave::PrivateCdnHelper::GetInstance()->RemovePadding(
-                 &body_payload))) {
-          // Byte padding removal failed
-          absl::optional<std::vector<uint8_t>> args;
-          std::move(callback).Run(std::move(args));
-          return;
-        }
-        // Download (and optional unpadding) was successful,
-        // uint8Array will be easier to move over mojom.
-        std::vector<uint8_t> image_bytes(body_payload.begin(),
-                                         body_payload.end());
-        std::move(callback).Run(image_bytes);
-      },
-      std::move(callback), is_padded);
-  api_request_helper_.Request("GET", padded_image_url, "", "", true,
-                              std::move(onPaddedImageResponse),
-                              brave::private_cdn_headers);
+  // Make the request
+  private_cdn_request_helper_.DownloadToString(
+      padded_image_url,
+      base::BindOnce(
+          [](GetImageDataCallback callback, const bool is_padded,
+             const int response_code, const std::string& body) {
+            // Handle the response
+            VLOG(3) << "getimagedata response code: " << response_code;
+            // Attempt to remove byte padding if applicable
+            base::StringPiece body_payload(body.data(), body.size());
+            if (response_code < 200 || response_code >= 300 ||
+                (is_padded &&
+                 !brave::PrivateCdnHelper::GetInstance()->RemovePadding(
```

Comment:
`RemovePadding` can be static.

Reply:
Would prefer to change in a follow-up. This is not introduced in this PR, and would prefer to limit this PR to fix the regression and not refactor.